### PR TITLE
Support `CommandRunner`/`Command`

### LINKED ISSUE #29: Support commands on a parsed class.
is there a way with build_cli to use addCommand ? if yes how ?

thanks

### PR DESCRIPTION
**Pull request by fzyzcjy**
Closes #29

### CODE REVIEW

**Comment by fzyzcjy** [comment]
ready for review now

**Comment by kevmoo** [comment]
Need to add this to build_cli/pubspec.yaml

```yaml

dependency_overrides:
  build_cli_annotations:
    path: ../build_cli_annotations
```

**Comment by fzyzcjy** [comment]
done

**Comment by kevmoo** [comment]
Yeah, we should have talked more about the output before you created everything.

I was thinking more about having options-within-options.

So if you have a nullable field of a class that annotated with `CliOptions` that would become a command.

Then maybe we do an extension on the class to point to which command is populated or something.

DM me on twitter. Let's set up a meeting and chat about this!

**Comment by fzyzcjy** [comment]
Sure, I have DM you :)

File: build_cli_annotations/lib/build_cli_annotations.dart [hunk 1]
```diff
@@ -1,7 +1,11 @@
 export 'package:args/args.dart' show ArgParser, ArgResults;
 
 class CliOptions {
-  const CliOptions();
+  final bool createCommand;
```
Comment (kevmoo) [inline build_cli_annotations/lib/build_cli_annotations.dart:4]: add a changelog entry and bump the pubspec version to 2.1.0

File: build_cli/pubspec.yaml [hunk 2]
```diff
@@ -30,3 +30,7 @@ dev_dependencies:
   path: ^1.0.0
   source_gen_test: ^1.0.0
   test: ^1.0.0
+
```
Comment (kevmoo) [inline build_cli/pubspec.yaml:33]: Change the `build_cli_annotations` constraint above to `'>=2.1.0 <2.2.0'`

Also change the version of `build_cli` to 2.2.0 – also add a changelog entry

**Review by kevmoo — CHANGES_REQUESTED** [review]

**Comment by fzyzcjy** [comment]
done

**Comment by kevmoo** [comment]
FYI: https://github.com/kevmoo/build_cli/pull/101

**Pull request merged**